c10/core/TensorTypeId.h

```diff
@@ -40,7 +40,11 @@ enum class TensorTypeId : uint8_t {
   ComplexCPUTensorId, // PyTorch only
   ComplexCUDATensorId, // PyTorch only
 
-  // VariableTensorId,  // upcoming!
+  // WARNING! If you add more non-backend-like tensor ids (like tracing or
```

Comment:
can we not use the word backend? :).

I admittedly don't have a better suggestion, but it seems the distinguishing characteristic is if you can write is a dispatch key in native_functions.yaml.  Speaking of which -- what happens if you try to do that now with Variable?

Reply:
I think it will work, actually. (Not very well; you'll end up with a double registration, probably, in that case.)